torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -788,6 +791,258 @@ def propagate_meta_info(new_hop_gm, new_call_function_node, old_call_function_no
     return joint_gm
 
 
+def maybe_log_graph(gm, graph_name, structured_log_tag, aot_config):
+    if not aot_config.enable_log:
+        return
+    aot_graphs_log.info(
+        "%s",
+        lazy_format_graph_code(
+            f"{graph_name}",
+            gm,
+            aot_config.aot_id,
+            include_stride=True,
+            include_device=True,
+            colored=True,
+        ),
+    )
+    gm_str = gm.print_readable(
+        print_output=False, include_stride=True, include_device=True
+    )
+    trace_structured(
+        f"{structured_log_tag}",
+        payload_fn=lambda: gm_str,
+    )
+
+
+def create_wrap_fn(fn, args):
+    from functools import wraps
+
+    from torch.fx.experimental.proxy_tensor import maybe_enable_thunkify
+
+    from .functional_utils import from_fun, has_data_mutation, to_fun
+
+    def assert_no_mutation(t):
+        assert not has_data_mutation(
+            t
+        ), "Saved tensors hooks with inputs mutations are not allowed"
+
+    @wraps(fn)
+    def _wrapper(*args):
+        with maybe_enable_thunkify():
+            disable_above = torch._C._ExcludeDispatchKeyGuard(
+                torch._C.DispatchKeySet(torch._C.DispatchKey.Functionalize)
+            )
+
+            with disable_above:
+                f_args = pytree.tree_map(to_fun, args)
+                f_outs = fn(*f_args)
+                pytree.tree_map(assert_no_mutation, f_args)
+                return pytree.tree_map(from_fun, f_outs)
+
+    return _wrapper, args
+
+
+def prepare_hook_gm(aot_config, fn, args):
+    from torch._functorch._aot_autograd.dispatch_and_compile_graph import _create_graph
+
+    fn, args = create_wrap_fn(fn, args)
+    gm = _create_graph(fn, args, aot_config=aot_config)
+    return gm
+
+
+# Inline Autograd saved_tensors_hooks into epilogue of forward graph
+# and prologue of backward graph.
+# This changes forward graph outputs and inputs.
+# Pack hook can return tensors, sym scalars, constants.
+# All tensors to save for backward will be grouped together at front.
+# Sym scalars grouped on another end. Constants are inlined in the graph.
+def maybe_inline_graph_saved_tensors_hooks(
+    fw_module, bw_module, num_inner_fwd_outputs, aot_config
+):
+    if torch._dynamo.compiled_autograd.in_compiled_autograd_region:
```

Comment:
why do we not support the inlining when using compiled autograd?

(compiled autograd will inline the unpack hooks properly, but the pack hooks will run in eager)

Reply:
if the answer is that we'll do this as a followup later then that seems fine :) just wanted to check if you thought there was something more fundamental